tokio/src/sync/oneshot.rs

```diff
@@ -544,6 +547,9 @@ impl<T> Inner<T> {
     }
 
     fn poll_recv(&self, cx: &mut Context<'_>) -> Poll<Result<T, RecvError>> {
+        // Keep track of task budget
+        ready!(crate::coop::poll_proceed(cx));
```

Comment:
Since a oneshot only completes once, I'm not sure if this needs to have a yield point? We can't exactly tie up a worker thread by continually receiving more and more messages on the same oneshot...

Reply:
In most cases it doesn't need to, but there are cases where it could help. For example, imagine something that has a large collection of oneshots, and then awaits all of them in a loop that does nothing else (like through `FuturesOrdered` or or something like that). The cost (should) be low, and it saves us from another case. The question, I suppose, is whether we feel like it's worth it.

Reply:
ahhh, yeah, that makes sense; i was thinking only within _this_ poll of the oneshot. this seems good to me!